File: frontend/src/types/market.ts
export interface Market {
  id: number;
  condition_id: string;
  poly_id: string | null;
  slug: string | null;
  question: string;
  description: string | null;
  category: string | null;
  image_url: string | null;
  event_title: string | null;
  end_date: string | null;
  active: boolean;

  // Prices
  yes_price: number | null;
  no_price: number | null;
  last_trade_price: number | null;
  one_day_price_change: number | null;
  best_bid: number | null;
  best_ask: number | null;
  spread: number | null;

  // Volume / liquidity
  volume: number | null;
  volume_24hr: number | null;
  liquidity: number | null;

  created_at: string;
  updated_at: string;
}

export interface MarketList {
  items: Market[];
  total: number;
}

export interface PriceUpdate {
  condition_id: string;
  yes_price: number | null;
  no_price: number | null;
  one_day_price_change: number | null;
  volume_24hr: number | null;
}

export type WSMessage =
  | { type: "snapshot"; markets: Market[]; total: number; timestamp: string }
  | { type: "price_update"; updates: PriceUpdate[]; timestamp: string }
  | { type: "pong" };

export type SortBy = "volume" | "volume_24hr" | "liquidity" | "end_date" | "price_change";
export type SortOrder = "asc" | "desc";

export interface MarketFilters {
  category: string | null;
  search: string;
  sortBy: SortBy;
  sortOrder: SortOrder;
}

File: backend/app/schemas/market.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class MarketRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    condition_id: str
    poly_id: str | None = None
    slug: str | None = None
    question: str
    description: str | None = None
    category: str | None = None
    image_url: str | None = None
    event_title: str | None = None
    end_date: datetime | None = None
    active: bool = True

    # Prices
    yes_price: float | None = None
    no_price: float | None = None
    last_trade_price: float | None = None
    one_day_price_change: float | None = None
    best_bid: float | None = None
    best_ask: float | None = None
    spread: float | None = None

    # Volume / liquidity
    volume: float | None = None
    volume_24hr: float | None = None
    liquidity: float | None = None

    created_at: datetime
    updated_at: datetime


class MarketList(BaseModel):
    items: list[MarketRead]
    total: int


class PriceUpdate(BaseModel):
    condition_id: str
    yes_price: float | None
    no_price: float | None
    one_day_price_change: float | None
    volume_24hr: float | None
